app/controllers/teachers/sessions_controller.rb

```diff
@@ -1,33 +1,48 @@
 # frozen_string_literal: true
-
 class Teachers::SessionsController < Devise::SessionsController
   include TeacherCurrentNamespace
 
   layout "two_thirds"
 
+  def new
+    @new_session_form = TeacherInterface::NewSessionForm.new
+  end
+
   def create
-    if resource_params[:create_or_sign_in] == "create"
-      redirect_to :eligibility_interface_countries
-      return
-    end
+    @new_session_form =
+      TeacherInterface::NewSessionForm.new(
+        email: new_session_form_params[:email],
+        create_or_sign_in: new_session_form_params[:create_or_sign_in]
+      )
 
-    self.resource = resource_class.find_by(email: resource_params[:email])
+    if @new_session_form.valid?
+      self.resource = resource_class.find_by(email: @new_session_form.email)
```

Comment:
We can move this line below the `if @new_session_form.create?` block as it's not used yet (saves an unnecessary database query).